renku/service/jobs/datasets.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataset jobs."""
+from urllib3.exceptions import HTTPError
+
+from renku.core.commands.dataset import import_dataset
+from renku.core.errors import ParameterError
+from renku.core.management.datasets import DownloadProgressCallback
+from renku.core.utils.contexts import chdir
+from renku.service.jobs.constants import USER_JOB_STATE_COMPLETED, \
+    USER_JOB_STATE_FAILED, USER_JOB_STATE_IN_PROGRESS
+from renku.service.serializers.cache import UserJob
```

Comment:
I'm just wondering about the abstractions here so feel free to ignore - but if `UserJob` is generic and used for different components (cache, datasets, others?) should it be in `serializers.cache` or maybe one level higher?